### ISSUE TITLE
Random crash caused by seafaring.

### ISSUE THREAD

**Issue by MarkMcWire** [widelands/widelands#4628]
widelands: ../src/economy/shipping_schedule.cc:570: void Widelands::ShippingSchedule::get_free_capacity_between(Widelands::Game&, const Widelands::Ship&, Widelands::ShipPlan&, Widelands::PortDock&, Widelands::PortDock&, bool&, bool&, bool&, bool&, Duration&, Duration&, uint32_t&): Zusicherung »ss.dock != &start« nicht erfüllt.

Occurs after an AI in the game defeats all of the other AIs. Endless game.

Savegame in https://www.widelands.org/forum/topic/5118/?page=1#post-35932

**Comment by Noordfrees** [comment]
I cannot reproduce the bug. When exactly does it happen? Is it reproducible? Any particular steps needed? Do you have a replay?

**Comment by MarkMcWire** [comment]
The bug is sporatic and cannot be triggered with secure. I have let run the game over night with 1x speed and after aprox. 5 hours I noticed that crash with this exception text in the terminal. (I started widelands from linux terminal via ./widelands).

I play since 11 am and no other crash happened in the meantime. 

It would be important to mention that I have set some ports of the AI myself in the game via debug script. Maybe this has caused the exception, but only after 5 hours delay. Only one AI player of two is active, myself and the other (Barbarians) are defeated.
